copy.c

```diff
@@ -1,11 +1,56 @@
+#include <fcntl.h>
 #include <stdio.h>
 #include <windows.h>
 
-int wmain(int argc, wchar_t **argv) {
-	DWORD cbSize = (wcslen(argv[1]) + 1) * 2;
+typedef struct node_tag {
+	char buffer[1024];
+	int capacity;
+	struct node_tag *next;
+} node;
+
+int wmain() {
+	_setmode(_fileno(stdin), _O_U8TEXT);
+
+	node *bufferList = malloc(sizeof(node));
+	if (bufferList == NULL) {
+		wprintf(L"Could not allocate memory for reading stdin.\n");
+		return 1;
+	}
+	bufferList->capacity = 0;
+	node *head = bufferList;
+
+	DWORD cbSize = 0;
+	DWORD count;
+	while ((count = fread(head->buffer, 1, sizeof(head->buffer), stdin)) != 0) {
+		cbSize += count;
+		head->capacity = count;
+		head->next = malloc(sizeof(node));
+		if (head->next == NULL) {
+			wprintf(L"Could not allocate memory for reading stdin.\n");
+			return 1;
+		}
+		head->next->capacity = 0;
+		head = head->next;
+	}
+
+	if (ferror(stdin)) {
+		fwprintf(stderr, L"Reading stdin failed.\r\n");
+		return 1;
+	}
+
 	HANDLE hgMem = GlobalAlloc(GMEM_MOVEABLE, cbSize);
```

Comment:
I tried searching GitHub for `GlobalAlloc` and it seems others are doing this too:
-
 https://github.com/karta0807913/ConsoleApplication1/blob/ee93c88aaef15dcdbd148e7e2cdd8cfc2c525bc7/ConsoleApplication1/SaveToClipboard.h#L10
-
 https://github.com/LasseSkogland/Scripts/blob/67828d6f90b272b2773004d9449c088557ba7b60/C%26Cpp%20Sources/Normalize/common.c#L9